pytorch_lightning/utilities/fetching.py

```diff
@@ -278,16 +241,14 @@ def fetching_function(self) -> Optional[Tuple[Any, bool]]:
             raise StopIteration
         else:
             try:
-                yield_batch = self.pop_batch()
+                yield_batch = self.batches.pop(0)
                 self._fetch_next_batch()
 
+                # TODO: move `wait` into `move_to_device`?
```

Comment:
Should this TODO be TODONE?

Reply:
I don't understand your question. I guess it won't matter since move_to_device is only used in two places and together with wait(). So either way should be good.

Reply:
Sorry, it was a bad joke. Yeah, the question was if it's worth moving it inside